src/main/java/org/gingesnap/k8s/CacheRuleWatcher.java

```diff
@@ -0,0 +1,61 @@
+package org.gingesnap.k8s;
+
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.Watch;
+import io.fabric8.kubernetes.client.Watcher;
+import io.fabric8.kubernetes.client.WatcherException;
+import io.quarkus.arc.lookup.LookupIfProperty;
+import io.quarkus.runtime.ShutdownEvent;
+import io.quarkus.runtime.StartupEvent;
+
+@Singleton
+public class CacheRuleWatcher {
+
+   private static final Logger log = LoggerFactory.getLogger(CacheRuleWatcher.class);
+   private static final String EAGER_CACHE_RULE_NAME = "EagerCacheRule";
```

Comment:
FYI the ConfigMap name(s) won't be constant and so will have to be a configurable property on startup.